web/src/components/core/FileViewer.jsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState, useEffect } from "react";
+import { Popup } from "~/components/core";
+import { Alert } from "@patternfly/react-core";
+import { LoadingEnvironment } from "~/components/layout";
+
+import cockpit from "../../lib/cockpit";
+
+import "./fileviewer.scss";
+
+// file loading indicator
+const spinner = () => {
+  return (
+    // set max. height to avoid resizing after the file is loaded (no "blinking")
+    <div style={{ height: "100vh" }}>
```

Comment:
Sorry for being nitpicker here, but this will not work as expected since the _spinner_ is a child of the _Popup_. Thus, the blinking will happen when instead of the file content you end up with an error (or simply when displaying a file with just a bunch of lines).

What is more, since you are using viewport units in a child… scroll will appear even when it's not needed.

Take a look at the screenshots below

| Spinner with scrollbars | The popup with an alert | The popup displaying small file |
|-|-|-|
|![Screenshot from 2023-02-06 11-52-48](https://user-images.githubusercontent.com/1691872/216964794-15c32164-7f28-4361-9692-5e02a6e0aeac.png) | ![Screenshot from 2023-02-06 11-54-15](https://user-images.githubusercontent.com/1691872/216965057-11280d0e-ca38-4053-81c2-23e081fb7cd6.png) | ![Screenshot from 2023-02-06 11-56-17](https://user-images.githubusercontent.com/1691872/216965448-ec670e66-a104-4d89-82e7-d94811a3d65c.png) |

Last but not least, I'd not use styles directly in the components unless strictly needed.

So, my suggestion if you want to fix this `Popup` _misbehavior_ here without touching other files, is to add a utility class "max-height", "full-height", "tallest", or so and use it with the popup. I.e., something like


```diff
diff --git a/web/src/assets/styles/utilities.scss b/web/src/assets/styles/utilities.scss
index bdbcd0c52..110b8901b 100644
--- a/web/src/assets/styles/utilities.scss
+++ b/web/src/assets/styles/utilities.scss
@@ -98,3 +98,10 @@
 .bottom-shadow {
   box-shadow: 0px -3px 10px 0px var(--color-gray-darker);
 }
+
+.tallest {
+  /** block-size fallbacks **/
+  height: 95dvh;
+  /** END block-size fallbacks **/
+  block-size: 95dvh;
+}
diff --git a/web/src/components/core/FileViewer.jsx b/web/src/components/core/FileViewer.jsx
index 7fe523429..837aea0cc 100644
--- a/web/src/components/core/FileViewer.jsx
+++ b/web/src/components/core/FileViewer.jsx
@@ -28,16 +28,6 @@ import cockpit from "../../lib/cockpit";
 
 import "./fileviewer.scss";
 
-// file loading indicator
-const spinner = () => {
-  return (
-    // set max. height to avoid resizing after the file is loaded (no "blinking")
-    <div style={{ height: "100vh" }}>
-      <LoadingEnvironment text="Reading file..." />
-    </div>
-  );
-};
-
 export default function FileViewer({ file, title, onCloseCallback }) {
   // the popup is visible
   const [isOpen, setIsOpen] = useState(true);
@@ -72,8 +62,9 @@ export default function FileViewer({ file, title, onCloseCallback }) {
       isOpen={isOpen}
       title={title || file}
       variant="large"
+      className="tallest"
     >
-      { state === "loading" && spinner() }
+      { state === "loading" && <LoadingEnvironment text="Reading file..." /> }
       { (content === null || error) &&
         <Alert
           isInline
```

Also remember that, whenever possible, we're trying to use the [CSS Logical Properties](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Logical_Properties) with minimal fallbacks that, hopefully, we could automate in a near future :)